src/ClientBuilderInterface.php

```diff
@@ -35,7 +42,7 @@ public static function create(array $options = []);
      *
      * @return $this
      */
-    public function setUriFactory(UriFactory $uriFactory);
+    public function setUriFactory(UriFactory $uriFactory): self;
```

Comment:
I explicitly avoided to use `self` in the interfaces because PHP doesn't have covariant returns and this forces anyone implementing this interface to define that the methods returns the interface and not the type of the class itself which can cause issues with fluent interfaces like this one. Please revert everything back